All_Relationships/9a.entity_prediction_russ_experiment.py

```diff
@@ -0,0 +1,518 @@
+from sqlalchemy import and_
+from sklearn.linear_model import LogisticRegression, SGDClassifier
+from sklearn.metrics import average_precision_score, precision_recall_curve, roc_curve, auc, f1_score, confusion_matrix
+from sklearn.model_selection import GridSearchCV
+from sklearn.preprocessing import StandardScaler
+import seaborn as sns
+
+
+# # Incorporate [Russ's Data](https://zenodo.org/record/1035500#.W3Hc-RgpBrk) into our Entity Dataframe
+
+# In[2]:
+
+
+named_columns = ["pubmed_id", "sentence_num", "first_entity", 
+                 "first_entity_location", "second_entity", "second_entity_location", 
+                 "first_entity_string", "second_entity_string", "first_entity_db_id", 
+                 "second_entity_db_id", "first_entity_type", "second_entity_type", 
+                 "dependency_path", "sentence_string"
+                ]
+russ_df = pd.read_table("data/hierarchical_clustering/part-ii-dependency-paths-gene-disease-sorted-with-themes.txt", 
+                        names=named_columns)
+russ_df.head(2)
+
+
+# In[3]:
+
+
+disease_gene_df = pd.read_csv("data/disease_gene_summary_stats.csv")
+disease_gene_df.head(2)
+
+
+# In[4]:
+
+
+disease_gene_russ_df = (
+    disease_gene_df[[col for col in disease_gene_df.columns if "lstm" not in col]]
+    .astype({'disease_name':str, 'gene_name':str})
+    .merge(
+        russ_df
+        .astype({'first_entity':str, 'second_entity':str}),
+        left_on=["gene_name", "disease_name"], right_on=["first_entity", "second_entity"], how='left')
+)
+(
+    disease_gene_russ_df[
+        ["disease_id", "disease_name", "gene_id", 
+         "gene_name", "nlog10_p_value", "co_odds_ratio", 
+         "co_expected_sen_count", "delta_lower_ci", "hetnet_labels", 
+         "pubmed_id", "sentence_num", "dependency_path", "sentence_string"]
+    ]
+    .head(2)
+)
+
+
+# In[5]:
+
+
+# fix the dependency path strings so they will match
+def fix(x):
+    x = re.sub('START_ENTITY', 'start_entity', x)
+    x = re.sub('END_ENTITY', 'end_entity', x)
+    return x
+
+
+# In[6]:
+
+
+disease_gene_russ_df.dependency_path = disease_gene_russ_df.dependency_path.astype(str).apply(fix)
+
+
+# In[7]:
+
+
+theme_dist_df = pd.read_table("data/hierarchical_clustering/part-i-gene-disease-path-theme-distributions.txt")
+theme_dist_df.head(2)
+
+
+# In[ ]:
+
+
+# fix the dependency path strings so they will match
+def fix(x):
+    x = re.sub('START_ENTITY', 'start_entity', x)
+    x = re.sub('END_ENTITY', 'end_entity', x)
+    return x
+
+
+# In[ ]:
+
+
+disease_gene_russ_df.dependency_path = disease_gene_russ_df.dependency_path.astype(str).apply(fix)
+
+
+# In[8]:
+
+
+final_dg_df = (
+    disease_gene_russ_df
+    .astype({'dependency_path':str})
+    .merge(theme_dist_df.astype({"path":str}), left_on='dependency_path', right_on='path', how='left')
+)
+final_dg_df.head(2)
+
+
+# In[9]:
+
+
+# Transform the nans into 0.0 and aggregate the other sentnece scores
+final_dg_df[['U', 'U.ind','Ud', 'Ud.ind',
```

Comment:
What does `.ind` mean?

Are these two lists of columns the same? If so, I'd just define them once as a variable above and use that twice. DRY.

Reply:
The .ind column is a binary variable that indicates if a dependency path is regarded as highly significant in the appropriate category. The naming was suppose to be intuitive, but I can change it if desired.